integration-cli/docker_cli_ps_test.go

```diff
@@ -666,3 +666,44 @@ func (s *DockerSuite) TestPsGroupPortRange(c *check.C) {
 	}
 
 }
+
+// Test for GitHub issue #12595
+func (s *DockerSuite) TestPsImageIDAfterUpdate(c *check.C) {
+
+	originalImageName := "busybox:TestPsImageIDAfterUpdate-original"
+	updatedImageName := "busybox:TestPsImageIDAfterUpdate-updated"
+
+	runCmd := exec.Command(dockerBinary, "tag", "busybox:latest", originalImageName)
+	out, _, err := runCommandWithOutput(runCmd)
+	c.Assert(err, check.IsNil)
+
+	originalImageID, err := getIDByName(originalImageName)
+	c.Assert(err, check.IsNil)
+
+	runCmd = exec.Command(dockerBinary, "run", "-d", originalImageName, "top")
+	out, _, err = runCommandWithOutput(runCmd)
+	c.Assert(err, check.IsNil)
+	containerID := strings.TrimSpace(out)
+
```

Comment:
Can you add a check here that ps will return with originalImageName, this makes the last check at the end more valuable.

I tried with this and it passed 
```diff
diff --git a/integration-cli/docker_cli_ps_test.go b/integration-cli/docker_cli_ps_test.go
index 80bd1db..8baa5a7 100644
--- a/integration-cli/docker_cli_ps_test.go
+++ b/integration-cli/docker_cli_ps_test.go
@@ -689,14 +689,27 @@ func (s *DockerSuite) TestPsImageIDAfterUpdate(c *check.C) {
        out, _, err = runCommandWithOutput(runCmd)
        c.Assert(err, check.IsNil)
 
+       linesOut, err := exec.Command(dockerBinary, "ps").CombinedOutput()
+       c.Assert(err, check.IsNil)
+
+       lines := strings.Split(strings.TrimSpace(string(linesOut)), "\n")
+       // skip header
+       lines = lines[1:]
+       c.Assert(len(lines), check.Equals, 1)
+
+       for _, line := range lines {
+               f := strings.Fields(line)
+               c.Assert(f[1], check.Equals, originalImageName)
+       }
+
        runCmd = exec.Command(dockerBinary, "tag", "-f", updatedImageName, originalImageName)
        out, _, err = runCommandWithOutput(runCmd)
        c.Assert(err, check.IsNil)
 
-       linesOut, err := exec.Command(dockerBinary, "ps").CombinedOutput()
+       linesOut, err = exec.Command(dockerBinary, "ps").CombinedOutput()
        c.Assert(err, check.IsNil)
 
-       lines := strings.Split(strings.TrimSpace(string(linesOut)), "\n")
+       lines = strings.Split(strings.TrimSpace(string(linesOut)), "\n")
        // skip header
        lines = lines[1:]
        c.Assert(len(lines), check.Equals, 1)
```